src/shared/tpm2-util.c

```diff
@@ -675,7 +675,7 @@ int tpm2_context_new(const char *device, Tpm2Context **ret_context) {
 
                 context->tcti_dl = dlopen(fn, RTLD_NOW);
                 if (!context->tcti_dl)
-                        return log_debug_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE), "Failed to load %s: %s", fn, dlerror());
+                        return log_debug_errno(SYNTHETIC_ERRNO(ENODEV), "Failed to load %s: %s", fn, dlerror());
```

Comment:
`ENODEV`? Shouldn't this use `EOPNOTSUP`? `ENODEV` seems error prone since it could just as well be used to indicate that no TPM device was found.

Reply:
I changed this to ENOPKG now.

I think EOPNOTSUPP would be worse, because that's what we support if the TPM exists and works, but doesn't support an operation we need. And there's value in showing a different message for that than for "TCTI driver not installed".

I think ENOPKG is a nice way out. it's different from both ENODEV and EOPNOTSUPP

Reply:
Sounds good to me